app/src/main/java/com/example/diceroller/MainActivity.kt

```diff
@@ -4,122 +4,80 @@ import android.os.Bundle
-            }
-
-            R.id.D6DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD6()
-            }
-
-            R.id.D8DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD8()
-            }
-
-            R.id.D10DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD10()
-            }
+    private fun setListeners() {
+        binding.menuButton.setOnClickListener { toggleNavMenu() }
+        binding.navView.setNavigationItemSelectedListener(this)
+    }
 
-            R.id.D12DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD12()
+    private fun toggleNavMenu() {
+        with(binding) {
+            when (navView.visibility) {
+                View.GONE -> navView.visibility = View.VISIBLE
+                View.VISIBLE -> navView.visibility = View.GONE
+                else -> navView.visibility = View.GONE
             }
+        }
+    }
 
-            R.id.D20DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD20()
-            }
+    override fun onNavigationItemSelected(selectedItem: MenuItem): Boolean {
 
-            R.id.LastRollsMenuItem -> {
-                navigateToFragment(fragmentContainer, listOfDiceFragment)
-            }
+        fun checkSelectedItem(selectedItem: MenuItem) {
+            val navigationMenu = binding.navView
 
-            else -> {
-                viewModel.triggerD6()
+            navigationMenu.menu.forEach {
+                it.isChecked = false
             }
+            selectedItem.isChecked = true
+        }
 
+        when (selectedItem.itemId) {
+            R.id.D4DiceMenuItem -> viewModel.onChangeDieIntent(DiceType.D4)
+            R.id.D6DiceMenuItem -> viewModel.onChangeDieIntent(DiceType.D6)
+            R.id.D8DiceMenuItem -> viewModel.onChangeDieIntent(DiceType.D8)
+            R.id.D10DiceMenuItem -> viewModel.onChangeDieIntent(DiceType.D10)
+            R.id.D12DiceMenuItem -> viewModel.onChangeDieIntent(DiceType.D12)
+            R.id.D20DiceMenuItem -> viewModel.onChangeDieIntent(DiceType.D20)
+            R.id.LastRollsMenuItem -> viewModel.onViewRollHistoryIntent()
+            else -> throw Exception("Unknown Navigation Item Selected")
         }
 
         checkSelectedItem(selectedItem)
-        navigationMenu.visibility = View.GONE
-
+        toggleNavMenu()
         return true
     }
-
-    private fun navigateToFragment(container: Int, fragment: Int) {
-        findNavController(container).navigate(fragment)
-    }
-
-    private fun checkSelectedItem(selectedItem: MenuItem) {
-        val navigationMenu = binding.navView
-        for (item in navigationMenu.menu) {
-            item.isChecked = false
-        }
```

Comment:
Whenever possible, use the kotlin stdlib functions. They will make your life a little easier. in this case i refactored this to use `forEach { }`